pkg/reconciler/pipelinerun/resources/pipelineref.go

```diff
@@ -146,26 +160,40 @@ func resolvePipeline(ctx context.Context, resolver remote.Resolver, name string,
 // An error is returned if the given object is not a
 // PipelineObject, trusted resources verification fails or if there is an error validating or upgrading an
 // older PipelineObject into its v1beta1 equivalent.
+// This method sets defaults and applies validation based on the API version of the referenced Pipeline.
 // TODO(#5541): convert v1beta1 obj to v1 once we use v1 as the stored version
 func readRuntimeObjectAsPipeline(ctx context.Context, obj runtime.Object, k8s kubernetes.Interface, refSource *v1beta1.RefSource, verificationPolicies []*v1alpha1.VerificationPolicy) (*v1beta1.Pipeline, error) {
+	// Inject a sentinel value into the context to skip validation of parameters and workspaces that may not yet be present
+	ctx = config.SkipValidationDueToPropagatedParametersAndWorkspaces(ctx, true)
 	switch obj := obj.(type) {
 	case *v1beta1.Pipeline:
 		// Verify the Pipeline once we fetch from the remote resolution, mutating, validation and conversion of the pipeline should happen after the verification, since signatures are based on the remote pipeline contents
-		err := trustedresources.VerifyPipeline(ctx, obj, k8s, refSource, verificationPolicies)
-		if err != nil {
-			return nil, fmt.Errorf("remote Pipeline verification failed for object %s: %w", obj.GetName(), err)
+		verificationErr := trustedresources.VerifyPipeline(ctx, obj, k8s, refSource, verificationPolicies)
+		if verificationErr != nil {
+			return nil, fmt.Errorf("remote Pipeline verification failed for object %s: %w", obj.GetName(), verificationErr)
+		}
+		obj.SetDefaults(ctx)
+		// Validation must happen before conversion to the stored version of the API, since validation may differ between API versions
+		validationErr := obj.Validate(ctx)
+		if validationErr != nil {
+			return nil, fmt.Errorf("remote Pipeline validation failed for object %s: %w", obj.GetName(), validationErr)
```

Comment:
```suggestion
		if err := obj.Validate(ctx); err != nil {
			return nil, fmt.Errorf("remote Pipeline validation failed for object %s: %w", obj.GetName(), err)
```
(here and elsewhere)

Scoping the error should help you avoid any naming collisions.